src/fsharp/PostInferenceChecks.fs

```diff
@@ -324,7 +324,7 @@ let rec CheckTypeDeep (cenv: cenv) ((visitTy,visitTyconRefOpt,visitAppTyOpt,visi
             | _ -> ())
     | _ -> ()
     
-    let ty = stripTyparEqns ty 
+    let ty = stripTyEqns g ty |> stripTyparEqns
```

Comment:
```suggestion
    let ty = 
        if isTyparTy g ty then
            stripTyparEqns ty
        else
            stripTyEqns g ty
```
The error in CI is caused by this. Perhaps combining both stripTyEqns and stripTyparEqns may not be wise to do.